test/bench/bench.cpp

```diff
@@ -132,11 +138,16 @@ void load_benchmark(const fs::path& path, const std::string& name_prefix)
     const auto wasm_binary = std::make_shared<const fizzy::bytes>(
         std::istreambuf_iterator<char>{wasm_file}, std::istreambuf_iterator<char>{});
 
-    register_benchmark("parse/" + base_name,
-        [wasm_binary](benchmark::State& state) { benchmark_parse(state, *wasm_binary); });
+    auto& engine = *g_engines[0];
```

Comment:
Also it doesn't feel very nice that all of the benchmarks operate on the same global data. Maybe better do a separate `create_fizzy_engine()` for each and move into lambdas?

Reply:
This will not scale with multiple engines. We need global registry of these. This is the simplest implementation, the global variable does not leave this file.

The change I'd like to make to it is to add factory functions (e.g. `create_fizzy_engine`) to it, instead of engine objects. The we can make the registry compile-time constant. Will that work for you?

Reply:
Yeah array of factory functions would be ok :+1: